### PR TITLE
oauth for pinterest

### PR DESCRIPTION
**Pull request by sudhnk**
(1) created oauth signin service for pinterest
(2) added pinterest method invokation in readme file

### CODE REVIEW

**Comment by nraboy** [comment]
Adding issue https://github.com/nraboy/ng-cordova-oauth/issues/199 and PR https://github.com/nraboy/ng-cordova-oauth/pull/201 for reference.

A few things here:

1. You are submitting to my master branch.  Only PR on the development branch will be approved.
2. Previous attempts at Pinterest have not gone well.  Have you tested this and confirmed that it is working without issues?

I'm adding @Slessi since he was part of the initial implementation.  I'm also adding @matheusrocha89 because I value his feedback when it comes to making this project a success.

Best,

File: src/oauth.pinterest.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            scope: appScope.join(","),
+            response_type: "code"
+          });
+          var flowUrl = "https://api.pinterest.com/oauth/?" + flowUrlParam;
+          var browserRef = window.cordova.InAppBrowser.open(flowUrl, '_blank', 'location=no,clearsessioncache=yes,clearcache=yes');
+          browserRef.addEventListener('loadstart', function (event) {
+            if ((event.url).indexOf(redirect_uri) === 0) {
+              var callbackResponse = (event.url).split("?")[1];
+              var responeObject = $cordovaOauthUtility.parseResponseParameters(callbackResponse);
+              if (responeObject.code !== undefined && responeObject.code !== null) {
```
Comment (matheusrocha89) [inline src/oauth.pinterest.js]: I think this could be just:
```javascript
if (responeObject.code)  { ... }
```

File: src/oauth.pinterest.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          var redirect_uri = "http://localhost/callback";
+          if (options !== undefined) {
+            if (options.hasOwnProperty("redirect_uri")) {
+              redirect_uri = options.redirect_uri;
+            }
+          }
+          var flowUrlParam = $cordovaOauthUtility.generateUrlParameters({
+            client_id: clientId,
+            redirect_uri: redirect_uri,
+            scope: appScope.join(","),
```
Comment (matheusrocha89) [inline src/oauth.pinterest.js]: Is this correct? The documentation said that the scope is separeted by `,` on URL request?

**Comment by matheusrocha89** [comment]
I maid some points on the code, besides them LGTM. Did you tested to see if it's working on iOS and Android? If yes, fine 😃  Just change this PR to development branch like @nraboy asked.

**Comment by Slessi** [comment]
As I recall the problems I encountered with Pinterest were the social login buttons they include in their OAuth page.

When I tried to use them it redirected me to Pinterest instead of back to my app upon authentication. Don't know if this is still the case, my use case was such that I just hid the social logins from the user with some injected CSS so I stopped testing further

**Comment by nraboy** [comment]
Thanks everyone for your feedback.

Let's wait for the contributor to respond.  If he doesn't respond within a week I will close the PR.

Best,

**Comment by nraboy** [comment]
No response from contributor, so I'm closing the PR.

**Pull request closed without merging**